compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -1124,3 +1125,27 @@ impl<'tcx> MutVisitor<'tcx> for Integrator<'_, 'tcx> {
         }
     }
 }
+
+#[instrument(skip(tcx), level = "debug")]
+fn try_instance_mir<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    instance: InstanceDef<'tcx>,
+) -> Result<&'tcx Body<'tcx>, &'static str> {
+    match instance {
+        ty::InstanceDef::DropGlue(_, Some(ty)) => match ty.kind() {
+            ty::Adt(def, substs) => {
```

Comment:
Not sure if we want to do this in a more generalized way. I believe the normalization in `move_paths_for_fields` is the only place where building drop shims can go wrong, so rejecting those types here should be sufficient (though this change feels a little hacky). 

Rejecting these here seemed preferable to me compared to making changes in the shim/drop elaboration code.